app/services/spree_shopify_importer/data_savers/zones/zone_base.rb

```diff
@@ -0,0 +1,74 @@
+module SpreeShopifyImporter
+  module DataSavers
+    module Zones
+      class ZoneBase < BaseDataSaver
+        delegate :attributes, to: :parser
+
+        private
+
+        def assign_spree_zone_to_data_feed
+          @shopify_data_feed.update!(spree_object: @spree_zone)
+        end
+
+        def parser
+          @parser ||= SpreeShopifyImporter::DataParsers::Zones::BaseData.new(shopify_shipping_zone)
+        end
+
+        def shopify_shipping_zone
+          @shopify_shipping_zone ||= ShopifyAPI::ShippingZone.new(data_feed)
+        end
+
+        def create_spree_zone_members
+          if attributes[:kind] == 'country'
+            find_countries_members
+          else
+            find_states_members
+          end
+        end
+
+        def find_countries_members
+          @shopify_shipping_zone.countries.map do |country|
```

Comment:
you have `map` here and below you have
```
Spree::Country.find_each do |country|
  create_country_member(country)
end
```

this find_each returns nil, and, as a result you'll have `nil` in your array

Reply:
it seems like you need `.each` here